### PR TITLE
feature: set link time optimization flag

### PR DESCRIPTION
**Pull request by Trafo**
mini-chromium part of crashpad builds by default with /GL and /LTCG flags under windows. But it is important that you can deactivate this flags. Because if this flags are activated, conan can not deal with them at the moment, because conan uses instead of the msvc compiler version the Visual Studio IDE version. If this flags are activated, crashpad needs to be build for every new msvc version, even if this is a minor version change. (This not happens under conan at the moment) Additionally this flags causes much more linking time, what during the developing phase is not needed.

### CODE REVIEW

**Review by reneme — APPROVED** [review]
LGTM

File: conanfile.py [hunk 1]
```diff
@@ -17,6 +17,8 @@ class CrashpadConan(ConanFile):
     url = "https://github.com/bincrafters/conan-crashpad"
     topics = ("conan", "crash-reporting", "logging", "minidump", "crash")
     settings = "os", "compiler", "build_type", "arch"
+    options = {'linktime_optimization': [True, False]}
+    default_options = "linktime_optimization=True"
```
Comment (reneme) [inline conanfile.py]: Given that conan cannot properly deal with the link time optimization flag: Should we disable this by default?

Comment (Trafo) [reply conanfile.py]: Yes I was thinking about this as well. But hopefully conan will fix it in the near future?

Comment (Trafo) [reply conanfile.py]: I changed it to False

File: conanfile.py [hunk 2]
```diff
@@ -17,6 +17,8 @@ class CrashpadConan(ConanFile):
     url = "https://github.com/bincrafters/conan-crashpad"
     topics = ("conan", "crash-reporting", "logging", "minidump", "crash")
     settings = "os", "compiler", "build_type", "arch"
+    options = {'linktime_optimization': [True, False]}
+    default_options = "linktime_optimization=False"
```
Comment (uilianries) [inline conanfile.py]:
```suggestion
    default_options = {"linktime_optimization": False}
```

File: conanfile.py [hunk 3]
```diff
@@ -59,12 +61,15 @@ def configure(self):
...
-                    patch_file="patches/dynamic_crt.patch")
-        # Backport of an upstream patch. Once crashpad gets updated to a version containing mini_chromium
-        # later than Aug 29, 2019, this patch can be removed again.
-        tools.patch(base_path=os.path.join(self._source_dir, "third_party/mini_chromium/mini_chromium"),
-                    patch_file="patches/fix_xcode11.patch")
+
+        if self.settings.os == "Windows":
+            tools.patch(base_path=os.path.join(self._source_dir, "third_party/mini_chromium/mini_chromium"),
+                        patch_file="patches/windows_adaptions.patch")
+        if self.settings.os == "Macos":
```
Comment (uilianries) [inline conanfile.py]:
```suggestion
        elif self.settings.os == "Macos":
```

**Review by uilianries — APPROVED** [review]

**Comment by Trafo** [comment]
@uilianries CI for macOS is green now.

**Pull request merged**